parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -175,12 +188,53 @@ pub mod pallet {
 	pub(super) type ExecutionHeaders<T: Config> =
 		StorageMap<_, Identity, H256, ExecutionHeader, OptionQuery>;
 
+	/// Execution headers ring buffer map implementation
+
+	/// Index storage for execution header ring buffer map
+	#[pallet::storage]
+	pub(crate) type ExecutionHeaderBufferIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+	/// Intermediate storage for execution header mapping
+	#[pallet::storage]
+	pub(crate) type ExecutionHeaderMapping<T: Config> =
+		StorageMap<_, Identity, u64, H256, ValueQuery>;
+
+	/// Ring buffer Map for Execution header
+	pub(crate) type ExecutionHeaderRingBufferMap<T> = RingBufferMapImpl<
+		u64,
+		<T as Config>::ExecutionHeadersPruneThreshold,
+		ExecutionHeaderBufferIndex<T>,
+		ExecutionHeaderMapping<T>,
+		ExecutionHeaders<T>,
+		OptionQuery,
+	>;
+
 	/// Current sync committee corresponding to the active header.
-	/// TODO  prune older sync committees than xxx
 	#[pallet::storage]
 	pub(super) type SyncCommittees<T: Config> =
 		StorageMap<_, Identity, u64, SyncCommitteeOf<T>, ValueQuery>;
 
+	/// Sync committee ring buffer map implementation
+
+	/// Index storage for sync committee ring buffer
+	#[pallet::storage]
+	pub(crate) type SyncCommitteesBufferIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+	/// Intermediate storage for sync committee mapping
+	#[pallet::storage]
+	pub(crate) type SyncCommitteesMapping<T: Config> =
+		StorageMap<_, Identity, u64, u64, ValueQuery>;
+
+	/// Ring buffer Map for Sync committee
+	pub(crate) type SyncCommitteesRingBufferMap<T> = RingBufferMapImpl<
```

Comment:
Lets reduce verbosity

```suggestion
	pub(crate) type SyncCommitteesIndex<T: Config> = StorageValue<_, u64, ValueQuery>;

	/// Intermediate storage for sync committee mapping
	#[pallet::storage]
	pub(crate) type SyncCommitteesMapping<T: Config> =
		StorageMap<_, Identity, u64, u64, ValueQuery>;

	/// Ring buffer Map for Sync committee
	pub(crate) type SyncCommitteesBuffer<T> = RingBufferMapImpl<
```

Reply:
@vgeddes Same as here: https://github.com/Snowfork/snowbridge/pull/815#discussion_r1180611132